tests/test_validator.py

```diff
@@ -0,0 +1,138 @@
+import unittest
+from click.testing import CliRunner
+from pathlib import Path
+
+from validator.__main__ import cli
+
+
+class TestXMLValidator(unittest.TestCase):
+    """Test for XML validator tool.
+
+    Testing the command line functions.
+    """
+    TESTFILES_ROOT = Path(__file__).parent / 'test_files'
+
+    def setUp(self):
+        """Initialise CLI runner and set paths to test files."""
+        self.runner = CliRunner()
+
+        # XML files and Schemas as published by ENA
+        self.xml_path = self.TESTFILES_ROOT / 'xml'
+        self.xsd_path = self.TESTFILES_ROOT / 'schemas'
+
+    def tearDown(self):
+        """Remove setup variables."""
+        pass
+
+    def test_no_args(self):
+        """Test case where no args are passed."""
+        result = self.runner.invoke(cli, [])
+        assert result.exception  # Catch any exception
```

Comment:
not sure this is needed to be checked.

Reply:
checking if any exception is raised or checking the no args case?

Reply:
then that is not the proper way you need: https://docs.python.org/3/library/unittest.html?highlight=assertcountequal#unittest.TestCase.assertRaises

Reply:
it is how they recommend here: https://click.palletsprojects.com/en/7.x/testing/ but you are using a unittest class, so use what is provided by that (as I mention: https://docs.python.org/3/library/unittest.html?highlight=assertcountequal#classes-and-functions)

Reply:
Yea, so I originally noticed that the assertRaises() method prints the CLI output as well, which made running the tests a bit messy. The latest commit now just checks the exit codes from each test which I imagine is conclusive enough?